tcf_website/views/browse.py

```diff
@@ -213,67 +214,66 @@ def course_instructor(request, course_id, instructor_id):
-            'avg_difficulty': safe_round(
-                Instructor.average_difficulty_for_course(
-                    instructor, course)),
-            'avg_gpa': safe_round(Instructor.average_gpa_for_course(instructor, course)),
-            'last_taught': str(course.semester_last_taught)}
-        grouped_courses[course_subdept]['courses'].append(course_data)
-
-    return grouped_courses
+    instructor: Instructor = get_object_or_404(Instructor, pk=instructor_id)
+    stats: Dict[str, float] = Instructor.objects\
+        .filter(pk=instructor_id)\
+        .prefetch_related('review_set')\
+        .aggregate(
+            avg_gpa=Avg('courseinstructorgrade__average'),
+            avg_difficulty=Avg('review__difficulty'),
+            avg_rating=(
+                Avg('review__instructor_rating') +
+                Avg('review__enjoyability') +
+                Avg('review__recommendability')
+            ) / 3)
+    course_fields: List[str] = ['name', 'id', 'avg_rating', 'avg_difficulty',
+                                'avg_gpa', 'last_taught']
+    queryset: List[Dict[str, Any]] = Course.objects\
+        .filter(section__instructors=instructor, number__gte=1000)\
+        .prefetch_related('review_set')\
+        .annotate(
+            dept=F('subdepartment__name'),
+            name=Concat(
+                F('subdepartment__mnemonic'),
+                Value(' '),
+                F('number'),
+                Value(' | '),
+                F('title'),
+                output_field=CharField(),
+            ),
+            avg_gpa=Avg('courseinstructorgrade__average',
+                        filter=Q(courseinstructorgrade__instructor=instructor)),
+            avg_difficulty=Avg('review__difficulty',
+                               filter=Q(review__instructor=instructor)),
+            avg_rating=(
+                Avg('review__instructor_rating',
+                    filter=Q(review__instructor=instructor)) +
+                Avg('review__enjoyability',
+                    filter=Q(review__instructor=instructor)) +
+                Avg('review__recommendability',
+                    filter=Q(review__instructor=instructor))
+            ) / 3,
+            last_taught=Concat(
+                F('semester_last_taught__season'),
+                Value(' '),
+                F('semester_last_taught__year'),
+                output_field=CharField(),
+            ),
+    ).values('dept', *course_fields)\
+        .order_by('dept', 'name')
+    courses: Dict[str, List[Dict[str, Any]]] = {}
+    for course in queryset:  # type: Dict[str, Any]
+        course['avg_rating'] = safe_round(course['avg_rating'])
+        course['avg_difficulty'] = safe_round(course['avg_difficulty'])
+        course['avg_gpa'] = safe_round(course['avg_gpa'])
+        course['last_taught'] = course['last_taught'].title()
+        courses.setdefault(course['dept'], []).append(course)
+    context: Dict[str, Any] = {
```

Comment:
I think this should also have whitespace before it

Reply:
Also fixed in 001b6592654e68ce1811a4652c6a21c4e4866f46